apps/studio/src/preload.ts

```diff
@@ -0,0 +1,209 @@
+import { contextBridge, ipcRenderer } from 'electron';
+import { AppEvent } from './common/AppEvent';
+import path from 'path';
+import fs, { readFileSync, WriteFileOptions, writeFileSync } from 'fs';
+import { Options } from 'yargs-parser';
+import yargs from 'yargs-parser';
+import { SettingsPlugin } from './plugins/SettingsPlugin';
+import { homedir } from 'os';
+import tls, { SecureVersion } from 'tls';
+import username from 'username';
+import { execSync } from 'child_process';
+import rawLog from 'electron-log/renderer';
+
+const log = rawLog.scope('preload.ts');
+
+const electron = require('@electron/remote');
+
+const isDevEnv = !(electron?.app && (electron?.app.isPackaged ?? process.env.isPackaged));
+const slice = isDevEnv ? 2 : 1;
+const parsedArgs = yargs(process.argv.slice(slice));
+// TODO: Automatically enable wayland without flags once
+// we're confident it will 'just work' for all Wayland users.
+function isWaylandMode() {
+  return parsedArgs['ozone-platform-hint'] === 'auto' &&
+    sessionType === 'wayland' && !isWindows && !isMac
+}
+
+const platform = process.env.OS_OVERRIDE ? process.env.OS_OVERRIDE : process.platform;
+const isWindows = platform === 'win32';
+const isMac = platform === 'darwin';
+const isArm = process.arch.startsWith('arm');
+const sessionType = process.env.XDG_SESSION_TYPE;
+const resourcesPath = isDevEnv ? path.resolve('./extra_resources') : path.resolve(process.resourcesPath);
+const testMode = process.env.TEST_MODE ? true : false;
+const easyPlatform = isWindows ? 'windows' : (isMac ? 'mac' : 'linux')
+const windowPrefersDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
+let userDirectory =  testMode ? './tmp' : electron?.app.getPath("userData")
+const downloadsDirectory = testMode ? './tmp' : electron?.app.getPath('downloads')
+const homeDirectory = testMode ? './tmp' : electron?.app.getPath('home')
+if (process.env.PORTABLE_EXECUTABLE_DIR) {
+  userDirectory = path.join(process.env.PORTABLE_EXECUTABLE_DIR, 'beekeeper_studio_data')
+}
+const updatesDisabled = !!process.env.BEEKEEPER_DISABLE_UPDATES
+const locale = electron?.app?.getLocale() ?? process.env.locale;
+
+function fileExistsSync(filename: string): boolean {
+  try {
+    return fs.statSync(filename).isFile();
+  } catch (e) {
+    return false;
+  }
+}
+
+
+export const api = {
+  isReady: () => {
+    ipcRenderer.send('ready');
+  },
+  send: (event: AppEvent, name: string, arg?: any) => {
+    if (!AppEvent[event]) return;
```

Comment:
This function would always stop here if we call it like `window.main.send(AppEvent.click, 'me')`.

Instead of `if(!AppEvent[event])`, maybe we can do `if(!_.hasValue(AppEvent, event))` or something like that.

Reply:
Good catch! I just wanna make sure we're at least doing some sort of whitelisting of events. Good suggestion!

Reply:
That's great! In that case, maybe adding a warning log in case we're calling a non-existing event will also be great.

Reply:
so `_.hasValue` doesn't seem to exist. Did some more digging online, and the best solution to me seems to be `if (!Object.values<string>(AppEvent).includes(event))`

Reply:
Yeah, I made that one up 😆 . Good find!